Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildTools.cs

```diff
@@ -17,11 +17,43 @@ namespace Microsoft.Build.Unity.ProjectGeneration
 {
     public class MSBuildToolsConfig
     {
+        private const int CurrentConfigVersion = 3;
+
```

Comment:
Is there a hope to have old configuration versions continue to work or is this just intended to make sure that things assessing the version know they are looking at an old configuration? If we want to support backwards compatibility, it may make more sense to keep the original MSBuildToolsConfig base class followed by MSBuildToolsConfig2, etc child classes with these new properties. You'll have to do runtime type inference to understand what version is supported, but you won't ever have fields that weren't originally populated.

Reply:
This is to ensure that if new fields are added and default  initialized, we immediately persist them after detecting the version change.

---
In reply to: [364490176](https://github.com/microsoft/MSBuildForUnity/pull/96#discussion_r364490176) [](ancestors = 364490176)